modules/core/src/main/scala/org/tessellation/infrastructure/rewards/StardustCollective.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.infrastructure.rewards
+
+import cats.data.NonEmptyMap
+
+import org.tessellation.config.types.StardustConfig
+import org.tessellation.domain.rewards.{NumberRefinementPredicatedFailure, RewardsError, StardustCollective}
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Amount
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+
+object StardustCollective {
+
+  def make(config: StardustConfig): StardustCollective = new StardustCollective {
+    def getAddress: Address = config.address
+
+    def weight(
```

Comment:
Why do you implicitly state 10L as a 100%? Imo it could be 0.1 with refinement 0-1. In case of 0-10 you'd have to set a refinement too imo. Also, why the remaining weight cannot be calculated? What if I'll set it 1 and 8?

Reply:
I think you're misreading this. These are weights, there's no assumption that `10L` is 100%. It could be `1L` and `2L` giving total of 3L and that would be a 100%. 

>Imo it could be 0.1 with refinement 0-1.

0.1 is not an integer number. We should do calculations only on integer numbers.

Reply:
Ok, got it :+1:

Reply:
So it's more like a fraction, not weight 🤔